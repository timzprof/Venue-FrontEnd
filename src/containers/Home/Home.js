import React, {Fragment, PureComponent} from "react";
import Header from "../../components/Header/Header";
import Footer from '../../components/Footer/Footer';
import VenueCard from "../../components/VenueCard/VenueCard";

class Home extends PureComponent {
	state = {
		venues: [
			{
				id: 1,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			},
			{
				id: 2,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			},
			{
				id: 3,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			},
			{
				id: 4,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			},
			{
				id: 5,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			},
			{
				id: 6,
				title: "Conference Hall",
				location: "CITS,Unilag",
				noOfSeats: 200,
				extras: [
					{
						type: "Internet",
						available: false
					},
					{
						type: "Computers",
						available: true
					}
				]
			}
		]
	};
	render() {
		return (
			<Fragment>
				<Header />
				<main className="main">
					<section className="card_container">
						{this.state.venues.map(venue => {
							return <VenueCard key={venue.id} {...venue} />;
						})}
					</section>
				</main>
				<Footer />
			</Fragment>
		);
	}
}

export default Home;
