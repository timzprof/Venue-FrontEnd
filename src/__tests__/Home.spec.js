import React from 'react';
import Home from '../containers/Home/Home';
import { create } from 'react-test-renderer';

test('Home Container renders', () => {
    const component = create(<Home />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})