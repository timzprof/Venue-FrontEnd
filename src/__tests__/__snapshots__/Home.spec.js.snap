// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container renders 1`] = `
Array [
  <header
    className="Header"
  >
    <div
      className="Logo"
    >
      VApp
    </div>
    <a
      className="Link"
      href="/login"
      onClick={[Function]}
    >
      Log In
    </a>
  </header>,
  <main
    className="main"
  >
    <section
      className="card_container"
    >
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
      <div
        className="Card Card"
      >
        <div
          className="Card_img"
        >
          <img
            alt="Venue"
            src="img.png"
          />
        </div>
        <h3>
          Conference Hall
        </h3>
        <p>
          CITS,Unilag
        </p>
        <p>
          200
           Seats
        </p>
        <button
          className="btn Card_button"
        >
          View
        </button>
      </div>
    </section>
  </main>,
  <footer>
    <p>
      Â© 2019 Copyright. Venue App.
    </p>
  </footer>,
]
`;
